src/Elastic.Markdown/Links/CrossLinks/CrossLinkFetcher.cs

```diff
@@ -87,6 +82,31 @@ protected async Task<LinkReference> Fetch(string repository, string[] keys, Canc
 		throw new Exception($"Repository found in link index however none of: '{string.Join(", ", keys)}' branches found");
 	}
 
+	// See https://github.com/elastic/docs-internal-workflows/issues/41
+	private async Task<Dictionary<string, LinkIndexEntry>> GetRepositoryLinksWithRetry(string repository, Cancel ctx)
+	{
+		const int maxAttempts = 3;
+		var attempts = 0;
+		while (true)
```

Comment:
Use the same attempts >= guard as when we throw?


While true loops should be reserved for long running operations